In [2]:
#imports
from pathlib import Path
import pandas as pd

In [3]:
# Define the path to the file
flight_data = Path("../Data/CSV/cleaned_flight_data.csv")

# Read the data from the file
flight_data_df = pd.read_csv(flight_data)
flight_data_df.head()

tbl  Year  quarter  departure_city_id  arrival_city_id  \
0  Table 1a  2010        1              34614            33195   
1  Table 1a  1998        4              30189            31703   
2  Table 1a  1998        4              30198            30852   
3  Table 1a  2009        3              32211            32575   
4  Table 1a  1993        4              30255            30852   

         departure_city                           arrival_city  \
0    Salt Lake City, UT          Tampa, FL (Metropolitan Area)   
1  Colorado Springs, CO  New York City, NY (Metropolitan Area)   
2        Pittsburgh, PA     Washington, DC (Metropolitan Area)   
3         Las Vegas, NV    Los Angeles, CA (Metropolitan Area)   
4        Huntsville, AL     Washington, DC (Metropolitan Area)   

   departure_airport_id  arrival_airport_id departure_airport  ... large_ms  \
0                 14869               15304               SLC  ...     0.38   
1                 11109               12197               COS  ...     0.73   
2                 14122               10821               PIT  ...     0.93   
3                 12889               14908               LAS  ...     0.58   
4                 12217               12264               HSV  ...     0.48   

   fare_lg                carrier_low  lf_ms fare_low                tbl1apk  \
0   247.69      US Airways Group Inc.   0.20   166.99  201011486915304SLCTPA   
1   292.60    Northwest Airlines Inc.   0.24   248.27  199841110912197COSHPN   
2   245.70  Continental Airlines Inc.   0.03    71.30  199841412210821PITBWI   
3   118.28      US Airways Group Inc.   0.41   102.75  200931288914908LASSNA   
4   275.50      United Air Lines Inc.   0.25   197.69  199341221712264HSVIAD   

   departure_latitude departure_longitude  arrival_latitude  arrival_longitude  
0           40.758478         -111.888142         37.860600         -78.804199  
1           38.835224         -104.819798         40.123164         -75.333718  
2           40.442169          -79.994945         38.892062         -77.019912  
3           36.169202         -115.140597         34.052238        -118.243344  
4           34.729538          -86.585283         38.892062         -77.019912  

[5 rows x 25 columns]

In [4]:
flight_data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 205189 entries, 0 to 205188
Data columns (total 25 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   tbl                   205189 non-null  object 
 1   Year                  205189 non-null  int64  
 2   quarter               205189 non-null  int64  
 3   departure_city_id     205189 non-null  int64  
 4   arrival_city_id       205189 non-null  int64  
 5   departure_city        205189 non-null  object 
 6   arrival_city          205189 non-null  object 
 7   departure_airport_id  205189 non-null  int64  
 8   arrival_airport_id    205189 non-null  int64  
 9   departure_airport     205189 non-null  object 
 10  arrival_airport       205189 non-null  object 
 11  nsmiles               205189 non-null  int64  
 12  passengers            205189 non-null  int64  
 13  fare                  205189 non-null  float64
 14  carrier_lg            205189 non-null  object 
 15  

In [5]:
flight_data_df.columns

Index(['tbl', 'Year', 'quarter', 'departure_city_id', 'arrival_city_id',
       'departure_city', 'arrival_city', 'departure_airport_id',
       'arrival_airport_id', 'departure_airport', 'arrival_airport', 'nsmiles',
       'passengers', 'fare', 'carrier_lg', 'large_ms', 'fare_lg',
       'carrier_low', 'lf_ms', 'fare_low', 'tbl1apk', 'departure_latitude',
       'departure_longitude', 'arrival_latitude', 'arrival_longitude'],
      dtype='object')

In [10]:
departure_airport = flight_data_df['departure_airport'].unique()
print(len(departure_airport))

178


In [12]:
arrival_airport =flight_data_df['arrival_airport'].unique()
print(len(arrival_airport))

163


In [13]:
departure_city = flight_data_df['departure_city'].unique()
print(len(departure_city))

136


In [15]:
arrival_city = flight_data_df['arrival_city'].unique()
print(len(arrival_city))

121


In [16]:
# Compute correlations
correlation_matrix = flight_data_df[['large_ms', 'fare_lg', 'lf_ms', 'fare_low']].corr()
print(correlation_matrix)

          large_ms   fare_lg     lf_ms  fare_low
large_ms  1.000000 -0.178920  0.521937 -0.091604
fare_lg  -0.178920  1.000000 -0.235000  0.839224
lf_ms     0.521937 -0.235000  1.000000  0.041644
fare_low -0.091604  0.839224  0.041644  1.000000


In [18]:
# Fixing the code
highest_fare = flight_data_df[['fare_lg', 'fare_low']].max()
lowest_fare = flight_data_df[['fare_lg', 'fare_low']].min()
average_fare = flight_data_df[['fare_lg', 'fare_low']].mean()

print("Highest Fares:\n", highest_fare)
print("Lowest Fares:\n", lowest_fare)
print("Average Fares:\n", average_fare)

Highest Fares:
 fare_lg     2725.6
fare_low    2725.6
dtype: float64
Lowest Fares:
 fare_lg     50.0
fare_low    50.0
dtype: float64
Average Fares:
 fare_lg     216.240639
fare_low    188.975523
dtype: float64


In [23]:
# Correlation between fare and passengers (demand)
correlation_fare_passengers = flight_data_df[['fare', 'passengers']].corr()

# Highest, lowest, and average fares
highest_fare = flight_data_df[['fare_lg', 'fare_low']].max()
lowest_fare = flight_data_df[['fare_lg', 'fare_low']].min()
average_fare = flight_data_df[['fare_lg', 'fare_low']].mean()

In [22]:
print("Correlation between fare and passengers:\n", correlation_fare_passengers)

Correlation between fare and passengers:
                fare  passengers
fare        1.00000    -0.19278
passengers -0.19278     1.00000


Correlation of -0.19278: This suggests a weak negative correlation between the fare and passengers variables. Specifically, as the average fare (fare) increases, the number of passengers (passengers) tends to decrease slightly. The closer the value is to -1, the stronger the negative correlation. In this case, the correlation of -0.19 is quite weak, meaning there isn’t a strong relationship between fare and passengers in your dataset.

In [25]:
print("Highest Fares:\n", highest_fare)
print("Lowest Fares:\n", lowest_fare)
print("Average Fares:\n", average_fare)

Highest Fares:
 fare_lg     2725.6
fare_low    2725.6
dtype: float64
Lowest Fares:
 fare_lg     50.0
fare_low    50.0
dtype: float64
Average Fares:
 fare_lg     216.240639
fare_low    188.975523
dtype: float64


The highest fares for both the largest and lowest carriers are the same (2725.6), which is unusual but could be a specific characteristic of the dataset.
	•	The lowest fares for both carriers are also identical (50.0).
	•	The average fares differ slightly, with the largest carrier charging on average 216.24, and the lowest fare carrier charging 188.98.

These statistics give you an overview of the fare distribution for both the largest and lowest carriers in your dataset.